tensorflow/lite/kernels/irfft2d_test.cc

```diff
@@ -0,0 +1,135 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <gtest/gtest.h>
+#include <stdint.h>
+
+#include <complex>
+#include <vector>
+
+#include "tensorflow/lite/interpreter.h"
+#include "tensorflow/lite/kernels/custom_ops_register.h"
+#include "tensorflow/lite/kernels/test_util.h"
+#include "tensorflow/lite/schema/schema_generated.h"
+#include "tensorflow/lite/testing/util.h"
+
+namespace tflite {
+namespace ops {
+namespace custom {
+
+namespace {
+
+using ::testing::ElementsAreArray;
+
+class Irfft2dOpModel : public SingleOpModel {
+ public:
+  Irfft2dOpModel(const TensorData& input, const TensorData& fft_lengths) {
+    input_ = AddInput(input);
+    fft_lengths_ = AddInput(fft_lengths);
+    TensorType output_type = TensorType_FLOAT32;
+    output_ = AddOutput({output_type, {}});
+
+    const std::vector<uint8_t> custom_option;
+    SetCustomOp("Irfft2d", custom_option, Register_IRFFT2D);
+    BuildInterpreter({GetShape(input_)});
+  }
+
+  int input() { return input_; }
+  int fft_lengths() { return fft_lengths_; }
+
+  std::vector<float> GetOutput() { return ExtractVector<float>(output_); }
+  std::vector<int> GetOutputShape() { return GetTensorShape(output_); }
+
+ private:
+  int input_;
+  int fft_lengths_;
+  int output_;
+};
+
+TEST(Irfft2dOpTest, FftLengthMatchesInputSize) {
+  Irfft2dOpModel model({TensorType_COMPLEX64, {4, 3}}, {TensorType_INT32, {2}});
+  // clang-format off
```

Comment:
Yes, I can. I didn't use clang_format because `rfft2d_test.cc` has the same coding style, i.e. disabling clang-format for these same sections. I copied that style for consistency.

If I change it, I would suggest to also change it in `rfft2d_test.cc`. However, I actually think the current style is more readable than with clang-format, and I assume the original author of `rfft2d_test.cc` did this for a good reason. I understand that readability is subjective to a certain extent, and I'm happy either way.